### PR TITLE
DolphinWX: Sync custom title changes on game list rescan

### PR DESCRIPTION
**Pull request by JosJuice**

### CODE REVIEW

File: Source/Core/DolphinWX/ISOFile.h [hunk 1]
```diff
@@ -33,7 +33,7 @@ class GameListItem
 {
 public:
   GameListItem() = default;
-  GameListItem(const std::string& file_name, const Core::TitleDatabase& title_database);
+  GameListItem(const std::string& file_name);
```
Comment (lioncash) [inline Source/Core/DolphinWX/ISOFile.h]: `explicit`

Comment (JosJuice) [inline Source/Core/DolphinWX/ISOFile.h]: Done.

File: Source/Core/DolphinWX/ISOFile.cpp [hunk 2]
```diff
@@ -147,6 +139,20 @@ bool GameListItem::IsValid() const
...
+  const auto type = m_platform == DiscIO::Platform::WII_WAD ?
+                        Core::TitleDatabase::TitleType::Channel :
+                        Core::TitleDatabase::TitleType::Other;
+  m_pending.custom_name = title_database.GetTitleName(m_game_id, type);
+  return m_custom_name != m_pending.custom_name;
+}
+
+void GameListItem::CustomNameCommit()
+{
+  m_custom_name = m_pending.custom_name;
```
Comment (shuffle2) [inline Source/Core/DolphinWX/ISOFile.cpp]: this could be a move (saves a bit of memory)

Comment (JosJuice) [inline Source/Core/DolphinWX/ISOFile.cpp]: I've made both this and `EmuStateCommit` use `move`.

**Pull request merged**